src/firefly/js/ui/DownloadDialog.jsx

```diff
@@ -0,0 +1,291 @@
+                                    ]}
+                                labelWidth = {labelWidth}
+                            />
+                        }
+                        <ListBoxInputField
+                            wrapperStyle={{marginTop: 5}}
+                            fieldKey ='zipType'
+                            initialState = {{
+                                        tooltip: 'Zip File Structure',
+                                        label : 'Zip File Structure:'
+                                    }}
+                            options = {[
+                                        {label: 'Structure (with folders)', value: 'folder'},
+                                        {label: 'Flattened (no folders)', value: 'flat'}
+                                    ]}
+                            labelWidth = {labelWidth}
+                        />
+                        <CheckboxGroupInputField
+                            wrapperStyle={{marginTop: 20}}
+                            fieldKey='sendEmail'
+                            initialState= {{label : ' '}}
+                            options={[
+                                        {label: 'Also send me email with URLs to download', value: 'true'}
+                                    ]}
+                            labelWidth = {0}
+                        />
+                        {sendEmail &&
+                            <div>
+                                <ValidationField
+                                    wrapperStyle={{marginTop: 5, marginLeft: 80}}
+                                    initialState = {{
+                                           value: '',
+                                           validator: Validate.validateEmail.bind(null, 'Email'),
+                                           label : 'Email:',
+                                           tooltip: 'Enter or modify your email address',
+                                       }}
+                                    fieldKey = 'Email'
+                                    size = {30}
+                                    labelWidth = {30}/>
+                            </div>
+                        }
+                    </FieldGroup>
+                </FormPanel>
+            </div>
+        );
+    }
+}
+
+
+DownloadOptionPanel.propTypes = {
+    groupKey:   PropTypes.string,
+    tbl_id:     PropTypes.string,
+    cutoutSize: PropTypes.string,
+    help_id:    PropTypes.string,
+    mask:       PropTypes.bool,
+    style:      PropTypes.object,
+    dlParams:   PropTypes.shape({
+        TitlePrefix:    PropTypes.string,
+        FilePrefix:     PropTypes.string,
+        BaseFileName:   PropTypes.string,
+        Title:          PropTypes.string,
+        DataSource:     PropTypes.string,
+        MaxBundleSize:  PropTypes.number,
+        FileGroupProcessor: PropTypes.string.isRequired,
+    })
+};
+
+DownloadOptionPanel.defaultProps= {
+    groupKey: 'DownloadDialog',
```

Comment:
Just noticed all these trailing commas. Looks like Trey finally moved you on his side. :)